From 4bfb5a9f63233628d831922167a31863199e026f Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Tue, 5 Oct 2021 17:59:16 -0400
Subject: [PATCH 01/33] Fallback to stupid autobrightness if brightness values
 are broken

This is needed because of:
`ava.lang.IllegalArgumentException: The control points must all have strictly increasing X values.`
on some Samsung devices

Change-Id: Ieb679b34239013a5e31b34cb010b12febd9ef6d9
---
 .../com/android/server/display/DisplayDeviceConfig.java    | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/display/DisplayDeviceConfig.java b/services/core/java/com/android/server/display/DisplayDeviceConfig.java
index 2ae5cbbbf24b..9cd8dc663cfa 100644
--- a/services/core/java/com/android/server/display/DisplayDeviceConfig.java
+++ b/services/core/java/com/android/server/display/DisplayDeviceConfig.java
@@ -172,7 +172,12 @@ public class DisplayDeviceConfig {
     public static DisplayDeviceConfig create(Context context, boolean useConfigXml) {
         DisplayDeviceConfig config;
         if (useConfigXml) {
-            config = getConfigFromGlobalXml(context);
+            try {
+                config = getConfigFromGlobalXml(context);
+            } catch(Exception e) {
+                android.util.Log.e("PHH", "Failed parsing automatic brightness values, fallbacking", e);
+                config = getConfigFromPmValues(context);
+            }
         } else {
             config = getConfigFromPmValues(context);
         }
-- 
2.32.0 (Apple Git-132)

